dll/win32/kernel32/client/dep.c

```diff
@@ -0,0 +1,147 @@
+/*
+ * PROJECT:         ReactOS system libraries
+ * LICENSE:         GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * FILE:            dll/win32/kernel32/client/dep.c
+ * PURPOSE:         ReactOS Data Execution Prevention (DEP) functions
+ * COPYRIGHT:       Copyright 2021 Oleg Dubinskiy (oleg.dubinskij2013@yandex.ua)
+ */
+
+/* INCLUDES *******************************************************************/
+
+#include <k32.h>
+
+#define NDEBUG
+#include <debug.h>
+
+/* PUBLIC FUNCTIONS ***********************************************************/
+
+/**
+ * @brief Retrieves the data execution prevention (DEP) and thunk emulation (DEP-ATL)
+ * settings for the process.
+ * 
+ * On Windows XP SP3, the current process is handled.
+ * On Vista and newer, the process specified in hProcess parameter is handled.
+ * 
+ * @param [in] hProcess
+ * A handle to a process with PROCESS_QUERY_INFORMATION privilege.
+ * Ignored on Windows XP SP3.
+ * 
+ * @param [out] lpFlags
+ * Pointer to a variable that receives the flags:
+ * 0 - DEP is disabled.
+ * PROCESS_DEP_ENABLE - DEP is enabled.
+ * PROCESS_DEP_DISABLE_ATL_THUNK_EMULATION - DEP-ATL is disabled.
+ * 
+ * @param [out] lpPermanent
+ * Pointer to a variable that specifies whether DEP is enabled or disabled permanently.
+ * 
+ * @return
+ * TRUE on success, FALSE on failure.
+ * 
+ * @remarks
+ * Supported for 32-bit processes only. Returns ERROR_NOT_SUPPORTED for 64-bit processes.
+ * 
+ */
+BOOL
+WINAPI
+GetProcessDEPPolicy(_In_ HANDLE hProcess,
+                    _Out_ LPDWORD lpFlags,
+                    _Out_ PBOOL lpPermanent)
+{
+    ULONG Flags;
+    NTSTATUS Status;
+
+    Status = NtQueryInformationProcess(GetCurrentProcess(),
```

Comment:
Why not use hProcess here?

Reply:
@learn-more see MSDN documentation:
> Gets the data execution prevention (DEP) and DEP-ATL thunk emulation settings for the specified 32-bit process.Windows XP with SP3:  Gets the DEP and DEP-ATL thunk emulation settings for the current process.

> [in] hProcess
A handle to the process. PROCESS_QUERY_INFORMATION privilege is required to get the DEP policy of a process.
Windows XP with SP3:  The hProcess parameter is ignored.

https://docs.microsoft.com/en-us/windows/win32/api/winbase/nf-winbase-getprocessdeppolicy

Reply:
Since these KERNEL32 API calls are public and exported, it's concise that you write some APITESTS so that you testify the behavior of these functions (in XP and in Server 2003 as well).

Reply:
Indeed, msdn is not always correct ;)